store/src/java/com/zimbra/cs/account/callback/EphemeralBackendCheck.java

```diff
@@ -76,12 +106,121 @@ private void savePreviousUrl(CallbackContext context, Entry entry) {
     @Override
     public void postModify(CallbackContext context, String attrName, Entry entry) {
         String prevUrl = context.getData(DataKey.PREV_EPHEMERAL_BACKEND_URL);
+        Config config = (Config) entry;
         try {
-            ((Config) entry).setPreviousEphemeralBackendURL(prevUrl);
+            config.setPreviousEphemeralBackendURL(prevUrl);
         } catch (ServiceException e) {
             ZimbraLog.ephemeral.error("unable to set zimbraPreviousEphemeralBackendURL to %s", prevUrl, e);
         }
+        try {
+            resetMigrationInfo(config.getEphemeralBackendURL());
+        } catch (ServiceException e) {
+            ZimbraLog.ephemeral.error("unable to reset attribute migration info", e);
+        }
         AttributeMigration.clearConfigCacheOnAllServers(false);
     }
 
+
+    /**
+     * Class encapsulating the rules governing whether the backend URL can be changed given
+     * what is known about the current state of ephemeral attribute migration
+     */
+    @VisibleForTesting
+    public static class EphemeralBackendMigrationRules {
+        private MigrationStateHelper helper;
+
+        public EphemeralBackendMigrationRules(MigrationStateHelper helper) {
+            this.helper = helper;
+        }
+
+
+        public void checkCanChangeURL(String newURL) throws ServiceException {
+            /*
+             * The only scenario that explicitly disallows changing the backend URL is if the migration is currently
+             * marked is IN_PROGRESS.
+             */
+            helper.setURL(newURL);
+            MigrationInfo info = AttributeMigration.getMigrationInfo();
+            Status curStatus = info.getStatus();
+            String migrationURL = info.getURL();
+            if (curStatus == Status.IN_PROGRESS) {
+                helper.deny();
+            } else if (curStatus == Status.NONE) {
+                helper.warn(Reason.NO_MIGRATION);
+            } else if (curStatus == Status.FAILED && newURL.equals(migrationURL)) {
```

Comment:
I was actually debating whether these cases should result in warnings or errors. An -f override is a good alternative, especially because running zmprov without the "-l" option doesn't actually output the warnings to the console but just hides them in the logs. I'll look into what it takes to propagate a new flag down to the attribute callbacks.